src/Symfony/Component/Console/Command/Command.php

```diff
@@ -134,9 +134,7 @@ public function __construct(?string $name = null)
             $this->setHelp($attribute?->help ?? '');
         }
 
-        if (\is_callable($this)) {
```

Comment:
I wonder if the fix instead could be to additionally check here that the `execute()` method is not overridden.

Reply:
In the example, `App\Console\Commands\HelloWorldCommand` doesn't have an `execute()` method, only `Illuminate\Console\Command` has it.

It would be nice if Symfony introduce an interface to handle invokable command (as this is a new feature for Symfony 7.3) so it doesn't rely on `is_callable()`.

Reply:
For the check I had in mind it doesn't really matter if it's an intermediate class or the actual command class that implements `execute()`. We would just ensure that the class declaring the method is not Symfony's `Command` class.

Reply:
> It would be nice if Symfony introduce an interface to handle invokable command (as this is a new feature for Symfony 7.3) so it doesn't rely on is_callable().

An interface is not desired on our side as we want commands to be any callable really.

Reply:
> In the example, `App\Console\Commands\HelloWorldCommand` doesn't have an execute() method, only `Illuminate\Console\Command` has it.

This confirm that the `execute()` method is overridden in `Illuminate\Console\Command`.

```php
if (\is_callable($this) && (new \ReflectionMethod($this, 'execute'))->getDeclaringClass() !== self::class) {
    $this->code = new InvokableCommand($this, $this(...));
}
```

@crynobone The solution you propose solves the issue you raise, but creates a new one because it introduces a new `protected` method tied to an implementation detail and becomes a maintenance constraint.

Reply:
@GromNaN I fine if that solve the issue.

Alternatively wouldn't it make more sense to resolve/invoke `$this->code` from `execute()` method, otherwise throw `LogicException`?

Reply:
> Alternatively wouldn't it make more sense to resolve/invoke $this->code from execute() method, otherwise throw LogicException?

That's a good idea, something I'd already thought of but had no good reason to do.

Reply:
I agree to add only ` && (new \ReflectionMethod($this, 'execute'))->getDeclaringClass()->name !== self::class` to this line, that does solve the original issue, right? please also add a new test around this case, thanks!

>Alternatively wouldn't it make more sense to resolve/invoke $this->code from execute() method, otherwise throw LogicException?

I don’t think it’s a good idea: invokable code detection should happen as early as possible due to its configuration in the `getNativeDefinition()` method (that’s why we perform this check in the constructor). Also, moving the `code` invocation to the `execute()` method will break the priority behavior, and thus affect some apps relying on `->setCode()` having higher priority over `execute()` (which is currently the case).

Reply:
My suggestion mainly moving changes from `run()` to `execute()`. I don't see how that would make a different.

Reply:
I opened https://github.com/symfony/symfony/pull/60361

The value of `$this->code` influence how the command definition is generated.

Reply:
Thanks @GromNaN